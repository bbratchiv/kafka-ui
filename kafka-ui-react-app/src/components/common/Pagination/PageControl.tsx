import React from 'react';

import { PaginationLink } from './Pagination.styled';

export interface PageControlProps {
  current: boolean;
  url: string;
  page: number;
}

const PageControl: React.FC<PageControlProps> = ({ current, url, page }) => {
  return (
    <li>
      <PaginationLink
        to={url}
        aria-label={`Goto page ${page}`}
        $isCurrent={current}
        data-testid="pagination-link"
      >
        {page}
      </PaginationLink>
    </li>
  );
};

export default PageControl;
