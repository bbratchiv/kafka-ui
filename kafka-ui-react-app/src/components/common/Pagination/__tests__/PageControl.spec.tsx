import React from 'react';
import { shallow } from 'enzyme';
import { StaticRouter } from 'react-router';
import PageControl, {
  PageControlProps,
} from 'components/common/Pagination/PageControl';
import { ThemeProvider } from 'styled-components';
import theme, { Colors } from 'theme/theme';
import { screen } from '@testing-library/react';
import { render } from 'lib/testHelpers';

const page = 138;

describe('PageControl', () => {
  const setupComponent = (props: Partial<PageControlProps> = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StaticRouter>
          <PageControl url="/test" page={page} current {...props} />
        </StaticRouter>
      </ThemeProvider>
    );

  it('renders current page', () => {
    setupComponent({ current: true });
    expect(screen.queryByTestId('pagination-link')).toHaveStyle(
      `background-color: ${Colors.neutral[10]}`
    );
  });

  it('renders non-current page', () => {
    setupComponent({ current: false });
    expect(screen.queryByTestId('pagination-link')).toHaveStyle(
      `background-color: ${Colors.neutral[0]}`
    );
  });

  it('renders page number', () => {
    setupComponent({ current: false });
    expect(screen.queryByTestId('pagination-link')).toHaveTextContent(
      String(page)
    );
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<PageControl url="/test" page={page} current />);
    expect(wrapper).toMatchSnapshot();
  });
});
