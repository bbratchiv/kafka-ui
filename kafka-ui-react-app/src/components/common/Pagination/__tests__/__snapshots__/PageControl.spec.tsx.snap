// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageControl matches snapshot 1`] = `
<li>
  <Styled(Link)
    $isCurrent={true}
    aria-label="Goto page 138"
    data-testid="pagination-link"
    to="/test"
  >
    138
  </Styled(Link)>
</li>
`;
